src/freenet/clients/fcp/FCPServer.java

```diff
@@ -454,6 +471,81 @@ private static String l10n(String key, String pattern, String value) {
 		return NodeL10n.getBase().getString("FcpServer."+key, pattern, value);
 	}
 
+    /**
+     * <p>Creates and registers {@link FCPPluginClient} object for FCP connections which are
+     * attached over the network.<br/>
+     * In other words, the actual client application is NOT a plugin running within the node, it
+     * only connected to the node via network.</p>
+     * 
+     * <p>The object is registered at the backend {@link FCPPluginClientTracker} and thus can be
+     * queried from this server by ID via the frontend {@link #getPluginClient(UUID)} as long as
+     * something else keeps a strong reference to it.<br/>
+     * Once it becomes weakly reachable, it will be garbage-collected from the backend
+     * {@link FCPPluginClientTracker} and {@link #getPluginClient(UUID)} will not work anymore.<br/>
+     * In other words, you don't have to take care of registering or unregistering clients. You only
+     * have to take care of keeping a strong reference to them while they are in use.</p>
+     * 
+     * <p>You should probably instead use the frontend function for this:
+     * {@link FCPConnectionHandler#getPluginClient(String)}.</p>
+     * 
+     * @see FCPPluginClient
+     *          The class JavaDoc of FCPPluginClient explains the code path for both networked and
+     *          non-networked FCP.
+     */
+    public FCPPluginClient createFCPPluginClientForNetworkedFCP(String serverPluginName,
+        FCPConnectionHandler messageHandler)
+            throws PluginNotFoundException {
+        
+        FCPPluginClient client = FCPPluginClient.constructForNetworkedFCP(node.executor,
+            node.pluginManager, serverPluginName, messageHandler);
+        pluginClientTracker.registerClient(client);
+        return client;
+    }
+
+    /**
+     * <p>Creates and registers a {@link FCPPluginClient} object for FCP connections between plugins
+     * running within the same node.<br/>
+     * In other words, the actual client application is NOT connected to the node by network, it is
+     * a plugin running within the node just like the server.</p>
+     * 
+     * <p>The object is registered at the backend {@link FCPPluginClientTracker} and thus can be
+     * queried from this server by ID via the frontend {@link #getPluginClient(UUID)} as long as
+     * something else keeps a strong reference to it.<br/>
+     * Once it becomes weakly reachable, it will be garbage-collected from the backend
+     * {@link FCPPluginClientTracker} and {@link #getPluginClient(UUID)} will not work anymore.<br/>
+     * In other words, you don't have to take care of registering or unregistering clients. You only
+     * have to take care of keeping a strong reference to them while they are in use.</p>
+     * 
+     * <p>You should probably instead use the frontend function:
+     * {@link PluginRespirator#connectToOtherPlugin(String,
+     * FredPluginFCPMessageHandler.ClientSideFCPMessageHandler)}.</p>
```

Comment:
Nice catch. Plugins should only use PluginRespirator, they shouldn't mess with the FCPServer.
Fixed by commit 8a9a952bab8b74fcd64f3aafc1362d3c01049edc